Refactor(eos_designs, eos_cli_config_gen): Adding new model ip_http_client and ip_ssh_client in place of ip_http_client_source_interfaces and ip_ssh_client_source_interfaces respectively (#6228)
aristanetworks/ansible-avd: docs/porting-guides/6.x.x.md

Reviewer: ```suggestion
In AVD 5.x, the data model `ip_telnet_client_source_interfaces` could generate multiple ip telnet client commands using the same source interface in different VRFs.
```

Reviewer: ```suggestion
In AVD 5.x, the data model `ip_tftp_client_source_interfaces` could generate multiple ip tftp client commands using the same source interface in different VRFs.
```

| `generate_default_config` | - |
| `generate_device_documentation` | `eos_cli_config_gen_documentation.enable` |
| `ip_name_server_groups[].ip_domain_list` | `ip_name_server_groups[].ip_domain_lists` |
| `ip_http_client_source_interfaces` | `ip_http_client` |
| `ip_ssh_client_source_interfaces` | `ip_ssh_client` |

> | `ip_ftp_client_source_interfaces` | `ip_ftp_client` |

Reviewer: let's try to keep the  alphabetical order sorting 
```suggestion
| `ip_ftp_client_source_interfaces` | `ip_ftp_client` |
| `ip_http_client_source_interfaces` | `ip_http_client` |
| `ip_name_server_groups[].ip_domain_list` | `ip_name_server_groups[].ip_domain_lists` |
| `ip_ssh_client_source_interfaces` | `ip_ssh_client` |
```

| `vlans[].tenant` | `vlans[].metadata.tenant` |
| `vrfs[].tenant` | `vrfs[].metadata.tenant` |

> ### IP HTTP client source interface data model changed from `ip_http_client_source_interfaces` to `ip_http_client`

Reviewer 2: I have created a separate section for vrfs data-model changes in my PR https://github.com/aristanetworks/avd/pull/6199

@gmuloc  should be keep the section or the separate entries like this are fine?

Reviewer: https://github.com/aristanetworks/avd/pull/6227/files#r2618673189

@Shivani-gslab lets use your section to group them together for clarify

Author: Updated

| Old data-model | New data-model |
| -------------- | -------------- |
| `ip_http_client_source_interfaces` | `ip_http_client` |
| `ip_ssh_client_source_interfaces` | `ip_ssh_client` |

> | `ip_ftp_client_source_interfaces` | `ip_ftp_client` |

Reviewer: ```suggestion
| `ip_ftp_client_source_interfaces` | `ip_ftp_client` |
| `ip_http_client_source_interfaces` | `ip_http_client` |
| `ip_ssh_client_source_interfaces` | `ip_ssh_client` |
```

| Old data-model | New data-model |
| -------------- | -------------- |
| `ip_ftp_client_source_interfaces` | `ip_ftp_client` |

> | `ip_http_client_source_interfaces` | `ip_http_client` |

Reviewer: let's order the sessions the same way as this table please

Author: Done

| `ip_ssh_client_source_interfaces` | `ip_ssh_client` |
| `ip_telnet_client_source_interfaces` | `ip_telnet_client` |
| `ip_tftp_client_source_interfaces` | `ip_tftp_client` |

> ### IP HTTP client source interface data model changed from `ip_http_client_source_interfaces` to `ip_http_client`

Reviewer: ```suggestion
#### IP HTTP client source interface data model changed from `ip_http_client_source_interfaces` to `ip_http_client`
```

In AVD 5.x, the data model `ip_http_client_source_interfaces` could generate multiple ip http client commands using the same source interface in different VRFs.

Starting with AVD 6.0.0, `ip_http_client_source_interfaces` has been replaced by `ip_http_client`, where only one unique ip http client command is rendered per VRF.

```diff
- ip_http_client_source_interfaces
-   - name: Management1
-   - name: Vlan4092
-     vrf: INBANDMGMT
+ ip_http_client:
+   source_interface: Management1
+   vrfs:
+     - name: INBANDMGMT
+       source_interface: Vlan4092
```

> ### IP SSH client source interface data model changed from `ip_ssh_client_source_interfaces` to `ip_ssh_client`

Reviewer: ``` suggestion
#### IP SSH client source interface data model changed from `ip_ssh_client_source_interfaces` to `ip_ssh_client`
```

| Old data-model | New data-model |
| -------------- | -------------- |
| `ip_ftp_client_source_interfaces` | `ip_ftp_client` |
| `ip_http_client_source_interfaces` | `ip_http_client` |
| `ip_ssh_client_source_interfaces` | `ip_ssh_client` |
| `ip_telnet_client_source_interfaces` | `ip_telnet_client` |
| `ip_tftp_client_source_interfaces` | `ip_tftp_client` |

### IP HTTP client source interface data model changed from `ip_http_client_source_interfaces` to `ip_http_client`

> In AVD 5.x, the data model `ip_http_client_source_interfaces` could generate multiple ip http client commands using the same VRF and source interface.

Reviewer: ugh so this is actually the correct wording... let's fix the other

```suggestion
In AVD 5.x, the data model `ip_http_client_source_interfaces` could generate multiple ip http client commands using the same source interface in different VRFs.
```

Starting with AVD 6.0.0, `ip_http_client_source_interfaces` has been replaced by `ip_http_client`, where only one unique ip http client command is rendered per VRF.

```diff
- ip_http_client_source_interfaces
-   - name: Management1
-   - name: Vlan4092
-     vrf: INBANDMGMT
+ ip_http_client:
+   source_interface: Management1
+   vrfs:
+     - source_interface: Vlan4092

> +       name: INBANDMGMT

Reviewer: we prefer with the primary_key first :)
```suggestion 
+     - name: INBANDMGMT
+       source_interface: Vlan4092
```

```

### IP SSH client source interface data model changed from `ip_ssh_client_source_interfaces` to `ip_ssh_client`

> In AVD 5.x, the data model `ip_ssh_client_source_interfaces` could generate multiple ip ssh client commands using the same VRF and source interface.

Reviewer: ```suggestion
In AVD 5.x, the data model `ip_ssh_client_source_interfaces` could generate multiple ip ssh client commands using the same source interface in different VRFs.
```

Starting with AVD 6.0.0, `ip_ssh_client_source_interfaces` has been replaced by `ip_ssh_client`, where only one unique ip ssh client command is rendered per VRF.

```diff
- ip_ssh_client_source_interfaces
-   - name: Management1
-   - name: Vlan4092
-     vrf: INBANDMGMT
+ ip_ssh_client:
+   source_interface: Management1
+   vrfs:
+     - source_interface: Vlan4092

> +       name: INBANDMGMT

Reviewer: ```suggestion
+     - name: INBANDMGMT
+       source_interface: Vlan4092
```

```

#### IP FTP client source interface data model changed from `ip_ftp_client_source_interfaces` to `ip_ftp_client`

> In AVD 5.x, the data model `ip_ftp_client_source_interfaces` could generate multiple ip ftp client commands using the same VRF and source interface.

Reviewer: ```suggestion
In AVD 5.x, the data model `ip_ftp_client_source_interfaces` could generate multiple ip ftp client commands using the same source interface in different VRFs.
```
